### PR TITLE
Vuln db

### PR DESCRIPTION
**Pull request by Shakedp**
## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/15044

## Description
Integration with VulnDB

## Required version of Demisto
5.0.0

## Does it break backward compatibility?
   - No

## Must have
- [ x ] Tests
- [ ] Documentation (with link to it) - will do after CR
- [ ] Code Review

### CODE REVIEW

**Comment by Itay4** [comment]
why is the required version here is 5.0 ?

File: Integrations/VulnDB/VulnDB.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+        'Description': vuln.get('description', '').rstrip('Z'),
+        'Keywords': vuln.get('keywords', ''),
+        'PublishedDate': vuln.get('vulndb_published_date', '').rstrip('Z'),
+        'TDescription': vuln.get('t_description', ''),
+        'SolutionDate': vuln.get('solution_date', '').rstrip('Z'),
+        'DiscoveryDate': vuln.get('disclosure_date', '').rstrip('Z'),
+        'ExploitPublishDate': vuln.get('exploit_publish_date', '').rstrip('Z'),
+    }
+
+    cve_ext_reference_values = [ext_reference['value'] for ext_reference in
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: Will we always have value? If not, please use `get`

File: Integrations/VulnDB/VulnDB.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        'Classification': classification_details
+    }
+
+
+def vulndb_vulnerability_results_to_demisto_results(res):
+    if 'error' in res:
+        return_error(res['error'])
+    else:
+        if 'vulnerability' in res:
+            results = [res['vulnerability']]
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: Is this a list and `result` below isn't necessarily a list on purpose?

File: Integrations/VulnDB/VulnDB.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        elif 'results' in res:
+            results = res['results']
+        else:
+            demisto.results({
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: `demisto.results()` doesn't `return`, it seems like you need to add a return after it

File: Integrations/VulnDB/VulnDB.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+                'HumanReadableFormat': formats['text']
+            })
+
+        for result in results:
+            ec = {
+                'VulnDB': vulndb_vulnerability_to_entry(result)
+            }
+
+            human_readable = tableToMarkdown(f'Result for vulnerability ID: {ec["VulnDB"]["Vulnerability"]["ID"]}', {
+                'Title': ec['VulnDB']['Vulnerability']['Title'],
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: please use `get`here and below

File: Integrations/VulnDB/VulnDB.py [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+        }
+    }
+
+
+def vulndb_vendor_results_to_demisto_results(res):
+    if 'error' in res:
+        return_error(res['error'])
+    else:
+        if 'vendor' in res:
+            results = [res['vendor']]
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: Is this a list and result below isn't necessarily a list on purpose?

File: Integrations/VulnDB/VulnDB.py [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+        elif 'results' in res:
+            results = res['results']
+        else:
+            demisto.results({
+                'Type': entryTypes['error'],
+                'Contents': res,
+                'ContentsFormat': formats['json'],
+                'HumanReadable': 'No "vendor" or "results" keys in the returned JSON',
+                'HumanReadableFormat': formats['text']
+            })
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: same with demisto.results and return

File: Integrations/VulnDB/VulnDB.py [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+
+
+def vulndb_product_results_to_demisto_results(res):
+    if 'error' in res:
+        return_error(res['error'])
+    else:
+        if 'results' in res:
+            results = res['results']
+        else:
+            demisto.results({
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: same like before with return

File: Integrations/VulnDB/VulnDB.py [hunk 8]
```diff
@@ -0,0 +1,431 @@
...
+
+        for result in results:
+            ec = {
+                'VulnDB': vulndb_vendor_to_entry(result)
+            }
+
+            human_readable = tableToMarkdown(f'Result for vendor ID: {ec["VulnDB"]["Results"]["Id"]}', {
+                'ID': ec['VulnDB']['Results']['Id'],
+                'Name': ec['VulnDB']['Results']['Name'],
+                'Short Name': ec['VulnDB']['Results']['ShortName'],
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: please use `get` here and below

File: Integrations/VulnDB/VulnDB.py [hunk 9]
```diff
@@ -0,0 +1,431 @@
...
+                'HumanReadableFormat': formats['text']
+            })
+
+        for result in results:
+            ec = {
+                'VulnDB': vulndb_product_to_entry(result)
+            }
+
+            human_readable = tableToMarkdown(f'Result for product ID: {ec["VulnDB"]["Results"]["Id"]}', {
+                'ID': ec['VulnDB']['Results']['Id'],
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: please use `get` here and below

File: Integrations/VulnDB/VulnDB.py [hunk 10]
```diff
@@ -0,0 +1,431 @@
...
+    os.environ.pop('https_proxy', None)
+
+CLIENT_ID = demisto.params()['client_id']
+
+CLIENT_SECRET = demisto.params()['client_secret']
+
+''' HELPER FUNCTIONS '''
+
+
+def get_oath_toekn():
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: should be `get_oath_token`

File: Integrations/VulnDB/VulnDB.py [hunk 11]
```diff
@@ -0,0 +1,431 @@
...
+    max_size = demisto.args().get('max_size')
+
+    if start_date:
+        url = f'{API_URL}/vulnerabilities/find_by_date?start_date={start_date}'
+        if end_date:
+            url += f'&end_date={end_date}'
+
+        res = http_request(url,
+                           max_size)
+    elif hours_ago:
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.py]: can we have `0` hours ago? If so we need to change the check

File: Integrations/VulnDB/VulnDB.yml [hunk 12]
```diff
@@ -0,0 +1,893 @@
...
+      secret: false
+    - default: false
+      description: specific vendor name (only human readable)
+      isArray: false
+      name: vendor_name
+      required: false
+      secret: false
+    - default: false
+      defaultValue: '20'
+      description: Maximum number of entries returned from the query, to avoid slow
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.yml]: Please go over this description (same for the other commands) with @kirbles19

File: Integrations/VulnDB/VulnDB.yml [hunk 13]
```diff
@@ -0,0 +1,893 @@
...
+        response.
+      isArray: false
+      name: max_size
```
Comment (hirshbergshachar) [inline Integrations/VulnDB/VulnDB.yml]: I think we should change this to `max_results_amount` or something like that, please consult with @kirbles19

**Comment by kirbles19** [comment]
Done editing the .yml.

**Pull request merged**